src/lib.rs

```diff
@@ -1,49 +1,69 @@
+#![deny(unsafe_code)]
+#![warn(clippy::cargo, clippy::missing_docs_in_private_items)]
+#![cfg_attr(doc, warn(rustdoc::all), allow(rustdoc::missing_doc_code_examples))]
+
+//! TODO
+
 use core::cmp::Ordering;
 
 use proc_macro2::{Ident, TokenStream};
-use proc_macro_error::{abort_call_site, proc_macro_error};
-use quote::{format_ident, quote};
+use quote::{format_ident, quote, ToTokens};
 use syn::parse::{Parse, ParseStream};
 use syn::punctuated::Punctuated;
 use syn::spanned::Spanned;
 use syn::{
-    parse_macro_input, Data, DeriveInput, Error, Fields, FieldsNamed, FieldsUnnamed, Token,
-    TraitBound, Type,
+    Data, DeriveInput, Error, Fields, FieldsNamed, FieldsUnnamed, Result, Token, TraitBound, Type,
 };
 
+/// Holds parsed [bounds](TraitBound) and [traits][Traits].
 struct DeriveWhere {
+    /// Parsed [bounds](TraitBound).
     bounds: Vec<TraitBound>,
+    /// Parsed [traits][Traits].
```

Comment:
Same here, I think both work just interested why

Reply:
I copied the typo from above and didn't notice, I would also like to know why this works 😄.

Reply:
I did a little bit of research, using square brackets for the link should point to a reference. Like this:
```markdown
[link text][link reference]

[link reference]: https://example.com
```
But somehow, it recognizes that the reference name is a link and just uses it, I guess there is some built-in leniency in the markdown parser. I didn't dig into Rusts markdown rules.